### PR TITLE
Misc. typos and whitespace fixes

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3 --skip="*.po,*.ts,./.git,./src/3rdParty,./src/CXX,./src/zipios++,./src/Mod/Assembly/App/opendcm" -I ../fc-word-whitelist.txt` and `grep`

### CODE REVIEW

File: src/Base/BaseClassPy.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
     </Methode>
     <Methode Name="getAllDerivedFrom" Const="true">
       <Documentation>
-        <UserDocu>Returns all descentences</UserDocu>
+        <UserDocu>Returns all descendence</UserDocu>
```
Comment (luzpaz) [inline src/Base/BaseClassPy.xml]: or should this be `descendants` ?

File: src/Mod/PartDesign/Gui/TaskFeaturePick.cpp [hunk 2]
```diff
@@ -58,6 +58,7 @@
 using namespace PartDesignGui;
 using namespace Attacher;
 
+// ***WMAYER, this comment needs revising***
 // TODO Do ve should snap here to App:Part or GeoFeatureGroup/DocumentObjectGroup ? (2015-09-04, Fat-Zer)
```
Comment (luzpaz) [inline src/Mod/PartDesign/Gui/TaskFeaturePick.cpp]: What should this comment say? (I'll remove line 61 once you answer)

File: src/Base/BaseClassPy.xml (hunk 1 above)
Comment (yorikvanhavre) [reply src/Base/BaseClassPy.xml]: yes, better descendents

File: src/Mod/PartDesign/Gui/TaskFeaturePick.cpp (hunk 2 above)
Comment (yorikvanhavre) [reply src/Mod/PartDesign/Gui/TaskFeaturePick.cpp]: leave it there for now, there are a lot of those in the source code, no way to solve everything at once

Comment (luzpaz) [reply src/Mod/PartDesign/Gui/TaskFeaturePick.cpp]: @yorikvanhavre I was referring to `TODO Do ve should snap` what does `Do ve` mean?

Comment (yorikvanhavre) [reply src/Mod/PartDesign/Gui/TaskFeaturePick.cpp]: not sure... Better to leave it like that

Comment (luzpaz) [reply src/Mod/PartDesign/Gui/TaskFeaturePick.cpp]: @wmayer any idea what Fat-Zer was trying to communicate ?

Comment (luzpaz) [reply src/Mod/PartDesign/Gui/TaskFeaturePick.cpp]: @yorikvanhavre the comment that I added needed to be wiped before merge :man_facepalming:

**Pull request merged**